### PR TITLE
[SYCL][InvokeSimd][E2E] Rework InvokeSimd SIMD32 tests

### PR DESCRIPTION
**Pull request by fveselov**
InvokeSimd/Spec/simd_size/simd32.cpp - enabled for all platforms and env added
InvokeSimd/Spec/simd_size/simd32_platform_error.cpp - removed

### CODE REVIEW

**Comment by dm-vodopyanov** [comment]
@fveselov please, follow https://github.com/intel/llvm/blob/sycl/sycl/doc/developer/ContributeToDPCPP.md#commit-message.

File: sycl/test-e2e/InvokeSimd/Spec/simd_size/simd32_platform_error.cpp [hunk 1]
```diff
@@ -1,30 +0,0 @@
-// Test not intended to run on PVC
-// UNSUPPORTED: gpu-intel-pvc
```
Comment (sarnex) [inline sycl/test-e2e/InvokeSimd/Spec/simd_size/simd32_platform_error.cpp:2]: can you explain why this test was removed? do we not want to error for simd32 on unsupported platforms anymore?

File: sycl/test-e2e/InvokeSimd/Spec/simd_size/simd32.cpp [hunk 2]
```diff
@@ -1,12 +1,9 @@
-// Test intended to run on PVC only
-// REQUIRES: gpu-intel-pvc
-//
 // Check that full compilation works:
 // RUN: %{build} -fno-sycl-device-code-split-esimd -Xclang -fsycl-allow-func-ptr -o %t.out
-// RUN: env IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %{run} %t.out
+// RUN: env IGC_ForceOCLSIMDWidth=32 IGC_VCSaveStackCallLinkage=1 IGC_VCDirectCallsOnly=1 %{run} %t.out
```
Comment (sarnex) [inline sycl/test-e2e/InvokeSimd/Spec/simd_size/simd32.cpp:3]: im not sure i understand why we need to pass this, shouldnt the subgroup size be determined to be 32 automatically, and the subgroup size has to equal the simd size?

File: sycl/test-e2e/InvokeSimd/Spec/simd_size/simd32_platform_error.cpp (hunk 1 above)
Comment (sarnex) [reply sycl/test-e2e/InvokeSimd/Spec/simd_size/simd32_platform_error.cpp:2]: discussed offline

**Pull request closed without merging**